### PR TITLE
oracledb - Add missing properties and undefined where neccessary

### PR DESCRIPTION
**Pull request by connorjayfitzgerald**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://oracle.github.io/node-oracledb/doc/api.html>>
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by Bigous — CHANGES_REQUESTED** [review]
It should be nice of yours to keep the original author and maintainer name in the comments.

**Review by Bigous — COMMENTED** [review]
@connorjayfitzgerald Thanks for your hard work and nice work on identifying optional items correctly.

File: types/oracledb/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // Type definitions for oracledb 4.0
 // Project: https://github.com/oracle/node-oracledb
-// Definitions by: Richard Natal <https://github.com/Bigous>
-//                 Connor Fitzgerald <https://github.com/connorjayfitzgerald>
+// Definitions by: Connor Fitzgerald <https://github.com/connorjayfitzgerald>
```
[the remark this answers is not in the document]
Comment (connorjayfitzgerald) [reply types/oracledb/index.d.ts]: Sorry about that @Bigous

[the remark this answers is not in the document]
Comment (connorjayfitzgerald) [reply types/oracledb/index.d.ts]: @Bigous Could you please approve the changes to progress this PR? Thanks.

[the remark this answers is not in the document]
Comment (Bigous) [reply types/oracledb/index.d.ts]: I've already sent the approval for the request... now it's the time to the bot of DefinitelyTyped to run and process everything

**Review by Bigous — COMMENTED** [review]
Everything ok

**Pull request closed without merging**